internal/internal_worker.go

```diff
@@ -1317,3 +1328,15 @@ func StartVersionMetrics(metricsScope tally.Scope) {
 		}()
 	})
 }
+
+// pollerLifeCycleImpl implements the PollerLifeCycle interface
+type pollerLifeCycleImpl struct{}
```

Comment:
nit: I suggest moving this into a separate package and providing an example of how customers can use it or how you expect to use it.